src/coreclr/jit/emitarm64.cpp

```diff
@@ -5676,6 +5676,22 @@ void emitter::emitIns_R_R_I(
         {
             return;
         }
+
+        if (emitComp->opts.OptimizationEnabled() && IsOptimisableLdrStr(ins, reg1, reg2, imm, size, fmt))
+        {
+            regNumber oldReg1 = emitLastIns->idReg1();
+            ssize_t   oldImm =
+                emitLastIns->idIsLargeCns() ? ((instrDescCns*)emitLastIns)->idcCnsVal : emitLastIns->idSmallCns();
+            instruction optIns       = (ins == INS_ldr) ? INS_ldp : INS_stp;
+            ssize_t     scaledOldImm = oldImm * size;
```

Comment:
It might be worth a comment here. I'm trying to understand why this isn't `+` or `* 2`.

Reply:
This line has been removed, due to changes to cater for register being stored to or loaded from both ascending an descending store location.
Nevertheless, a comment has been added regarding the scaling of the apprpriate immediate value.